Source/Samples/109_KinematicCharacter/KinematicCharacterDemo.cpp

```diff
@@ -266,63 +271,17 @@ void KinematicCharacterDemo::Update(float timeStep)
 
     if (character_)
     {
-        // Clear previous controls
-        character_->controls_.Set(CTRL_FORWARD | CTRL_BACK | CTRL_LEFT | CTRL_RIGHT | CTRL_JUMP, false);
-
-        // Update controls using touch utility class
-        if (touch_)
-            touch_->UpdateTouches(character_->controls_);
-
         // Update controls using keys
         auto* ui = GetSubsystem<UI>();
         if (!ui->GetFocusElement())
         {
-            if (!touch_ || !touch_->useGyroscope_)
```

Comment:
Why remove existing features?

Reply:
This code was only used in Character demo and KinematicCharacter which is the copy of first. I guess it was written at time when gyroscope was considered as an option. Can you name a modern TPS/FPS like game where you control character with gyro?

Reply:
Aiming with gyro is terrific on steam controller. But im talking more about example of touch controls. Having those is very useful for viewing demos on mobiles.

Reply:
I see component itself has touch support, which is great! This would make this comment of mine not relevant as there is no feature loss. What about on-screen controls though? If they could also be provided by the component it would be amazing!

Reply:
On screen buttons are also handled. You can bind on screen button to a key by name and then query everything at once.